[RNMobile] Block title in unsupported block

hypest opened the pull request:
(this is a follow up from #17943 and will copy and amend its description here. Recreated that PR so we merge it while @illusaen is temporarily away)

## Description
* Changed text on `Unsupported` block to show original block name.
* Added a help icon that shows additional information when the block is clicked.
* Append "Unsupported" to the title of all unsupported blocks

## How has this been tested?
* Tested on simulator and on Android device.
* Wrote unit tests for component change.
* Component change affects all types of blocks that were defaulting to the 'Missing' block (aka `UnsupportedBlockEdit`)

## Screenshots
![unsupported](https://user-images.githubusercontent.com/13263478/66786237-b4b4a000-eead-11e9-9043-3ac47279c4e4.gif)

## Types of changes
* New feature: [Issue #479](https://github.com/wordpress-mobile/gutenberg-mobile/issues/479)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->

illusaen committed:
WC: #479: Adding help icon and modal to unsupported block. Also putting original name in place of 'Unsupported'.

illusaen committed:
WC: #479: Adding in unit tests for the component.

illusaen committed:
WC: #479: Changing the help icon press surface to the whole block because the small icon was too hard to press. Also rearranging the BottomSheet to work the height paddings out correctly.

illusaen committed:
WC: #479: Changing accessibility label and adding comment.

illusaen committed:
WC: #479: Adding in text/buttons to match later design.

hypest committed:
Display 'unsupported' on all unsupported blocks on mobile

illusaen committed:
WC: issue #479: Changing only the question mark icon to be tappable for the help block. Few color changes. Adding some text to the help block.

hypest committed:
Merge branch 'master' into rnmobile/block-title-in-unsupported-block

hypest committed:
Update the test snapshot of Unsupported block on mobile

hypest committed:
Need to mock styling on mobile

hypest committed:
Adjust tests to latest changes

hypest committed:
Display the block name in quotes, for clarity

maxme commented on packages/block-library/src/missing/style.native.scss:
```diff
@@ -1,7 +1,72 @@
+	text-align: center;
+}
+
+.infoTitle {
```
I think we must create a `infoTitleDark` class and update the color.

hypest replied:
Addressed with 5a9e429.

maxme commented on packages/block-library/src/missing/style.native.scss:
```diff
@@ -1,7 +1,72 @@
+	color: $gray-dark;
+}
+
+.infoDescription {
```
Same as for `infoTitle`, we should create a `infoDescriptionDark` class and update the color.

hypest replied:
Addressed with 5a9e429.

maxme commented on packages/block-library/src/missing/edit.native.js:
```diff
@@ -18,26 +18,86 @@ import { __ } from '@wordpress/i18n';
+		>
+			<View style={ styles.infoContainer } >
+				<Icon icon="editor-help" style={ styles.infoIcon } size={ styles.infoIcon.size } />
+				<Text style={ [ styles.infoText, styles.infoTitle ] }>
```
We should call `getStylesFromColorScheme` on there.

hypest replied:
Addressed with 5a9e429.

maxme reviewed (changes requested):
Current implementation is off with dark mode for iOS, there are some changes needed in the style sheet and the block implementation (see review comments).

![Simulator Screen Shot - iPhone 11 - 2019-11-04 at 16 48 15](https://user-images.githubusercontent.com/40213/68135206-975e7900-ff23-11e9-8afe-b2698f3870e7.png)

koke commented on packages/block-library/src/missing/edit.native.js:
```diff
@@ -18,26 +18,86 @@ import { __ } from '@wordpress/i18n';
+			<View style={ styles.infoContainer } >
+				<Icon icon="editor-help" style={ styles.infoIcon } size={ styles.infoIcon.size } />
+				<Text style={ [ styles.infoText, styles.infoTitle ] }>
+					{ __( '\'' + title + '\' isn\'t yet supported on WordPress for ' + platformText ) }
```
You can't compose strings inside `__()`, you need to use `sprintf` instead and add a placeholder for the title. For the platform, I think it's safer to have one string for iOS and one for Android, since it's more obvious for translators.

hypest replied:
Addressed with 4c2b904.

koke commented on packages/block-library/src/missing/edit.native.js:
```diff
@@ -18,26 +18,86 @@ import { __ } from '@wordpress/i18n';
+	}
+
+	renderSheet( title ) {
+		return <BottomSheet
```
I'm surprised that it's not caught by the lint job, but I noticed a different JSX style in this file than the rest of the code: we normally wrap the returned JSX in parentheses and jump to a new line:

```jsx
return (
    <BottomSheet
        isVisible={ this.state.showHelp }
 	hideHeader
        onClose={ this.toggleSheet.bind( this ) }
    >
        { /* ... */ }
    </BottomSheet>
);
```

I can't find it mentioned in any coding style guide, but that's the style I've seen and written everywhere else.

hypest replied:
Addressed in f7c8582.

hypest committed:
Dark mode of info text, title and description

hypest committed:
sprintf to compose string, separate strings per platform

hypest committed:
Wrap JSX in the grouping operator, as usual

hypest committed:
Remove dismiss button, not used in other bottom-sheets

hypest committed:
Remove unused import

hypest committed:
Revise sprintf format to green the linter

hypest commented:
👋 @maxme and @koke , thanks for the review! I've addressed your comments and this is ready for another pass.

I've also removed the "Close" button from the bottom sheet as we don't have such a dismiss button in other bottom sheets anyway. cc @iamthomasbishop, there was a [relevant question in the original PR](https://github.com/WordPress/gutenberg/pull/17943#issuecomment-542142413) and I went with a "No" here... let me know if you think we should reinstate the Close button.

hypest committed:
Update tests after removing the Close button

hypest committed:
Merge branch 'master' into rnmobile/block-title-in-unsupported-block

koke commented on packages/block-library/src/missing/edit.native.js:
```diff
@@ -1,43 +1,106 @@
+	}
+
+	renderHelpIcon() {
+		return <TouchableWithoutFeedback
```
There are also parentheses missing here. Also FYI, suggested this as a lint rule in #18302

hypest replied:
Addressed with e863087.

koke commented on packages/block-library/src/missing/edit.native.js:
```diff
@@ -1,43 +1,106 @@
+
+		// translators: %s: Name of the block
+		const titleFormat = Platform.OS === 'android' ? __( '\'%s\' isn\'t yet supported on WordPress for Android' ) :
+			__( '\'%s\' isn\'t yet supported on WordPress for iOS' );
```
Not an issue for this PR, but noting for the future:  I just realized this might be used in other apps, so we should find  a way to inject the app name from the apps instead.

hypest replied:
Good point!

hypest committed:
Merge branch 'master' into rnmobile/block-title-in-unsupported-block

hypest committed:
Wrap more JSX in the grouping operator, as usual

hypest commented:
Addressed the additional feedback @koke , ready for another pass.

maxme reviewed (approved):
I can't find color definition on the original issue (https://github.com/wordpress-mobile/gutenberg-mobile/issues/479) or on the other PR (#17943). Anyway, the end results looks good to me. If the color scheme have to be changed, it can be done in a follow up PR.

:shipit: 

![Simulator Screen Shot - iPhone 11 - 2019-11-06 at 18 04 11](https://user-images.githubusercontent.com/40213/68320389-2902ed80-00c0-11ea-8dab-a1d6a3e87c47.png)

iamthomasbishop commented:
@maxme Looks good – the only color change I'd request is to change the `(?)` icons on the sheet and on the block itself to [gray 20](http://color-studio.blog).

hypest committed:
Adjust the info-icon color

hypest commented:
> the only color change I'd request is to change the (?) icons on the sheet and on the block itself to gray 20 (in dark mode).

Addressed with 92d7e71.

I set the light-mode icon color to `gray-darken-20`. Can you verify @iamthomasbishop that that's the light-mode color we want? Thanks! 

Screenshots:

Light-mode list | Dark-mode list | Light-mode popup | Dark-mode popup
--- | --- | --- | ---
![Simulator Screen Shot - iPhone X - 2019-11-07 at 10 31 21](https://user-images.githubusercontent.com/1032913/68372540-c3f3da00-0149-11ea-92e6-55a88f47a4d8.png) | ![Simulator Screen Shot - iPhone X - 2019-11-07 at 10 31 03](https://user-images.githubusercontent.com/1032913/68372573-d40bb980-0149-11ea-8afc-d7a557632636.png) | ![Simulator Screen Shot - iPhone X - 2019-11-07 at 10 31 18](https://user-images.githubusercontent.com/1032913/68372603-e2f26c00-0149-11ea-883b-dceecabd3a0a.png) | ![Simulator Screen Shot - iPhone X - 2019-11-07 at 10 31 07](https://user-images.githubusercontent.com/1032913/68372624-eb4aa700-0149-11ea-9441-f313581db332.png)

hypest committed:
Merge branch 'master' into rnmobile/block-title-in-unsupported-block

hypest committed:
Merge branch 'master' into rnmobile/block-title-in-unsupported-block

hypest commented:
This PR already has a 👍 and checked in with @koke too and he's fine with the PR state. The colors should be OK now @iamthomasbishop, but if they're not we can open a new PR really quickly, no problem. I'll merge this one as soon as the gutenberg-mobile side PR also gets the 👍.

hypest committed:
Merge branch 'master' into rnmobile/block-title-in-unsupported-block

SergioEstevao reviewed (approved):
Working great on iOS (dark and light) and Android.

iamthomasbishop commented:
@hypest Oops, I meant that `gray 20` should be applied only to _dark mode_. I realize now my comment wasn't the clearest, so apologies for that 😄 That initial "old gray" is fine on light mode.

hypest commented:
Sorry for the confusion @iamthomasbishop , apparently I should have restarted the app inbetween mode switching for the proper colors to appear. Without changing the code, here are new screenshots taken:

Light-mode list | Dark-mode list | Light-mode popup | Dark-mode popup
--- | --- | --- | ---
![Simulator Screen Shot - iPhone X - 2019-11-07 at 17 47 19](https://user-images.githubusercontent.com/1032913/68404339-0be52200-0187-11ea-968b-ced8e61b89d5.png) | ![Simulator Screen Shot - iPhone X - 2019-11-07 at 17 48 23](https://user-images.githubusercontent.com/1032913/68404362-169fb700-0187-11ea-8519-b961e88d9277.png) | ![Simulator Screen Shot - iPhone X - 2019-11-07 at 17 47 27](https://user-images.githubusercontent.com/1032913/68404377-1e5f5b80-0187-11ea-9480-a9a961f7cb5c.png) | ![Simulator Screen Shot - iPhone X - 2019-11-07 at 17 48 27](https://user-images.githubusercontent.com/1032913/68404392-24553c80-0187-11ea-856c-d4e7b3b773d5.png)

hypest committed:
Different colors for the icon on the sheet

hypest commented:
After chatting with @iamthomasbishop over Slack, revised the light-mode color of the icon on the sheet to be `gray`. Screenshots:

Light-mode list | Dark-mode list | Light-mode popup | Dark-mode popup
--- | --- | --- | ---
![Simulator Screen Shot - iPhone X - 2019-11-07 at 18 08 16](https://user-images.githubusercontent.com/1032913/68406062-cfff8c00-0189-11ea-9c9c-2ea02389c9b1.png) | ![Simulator Screen Shot - iPhone X - 2019-11-07 at 18 09 44](https://user-images.githubusercontent.com/1032913/68406082-d7269a00-0189-11ea-93d0-f09ac1e9917e.png) | ![Simulator Screen Shot - iPhone X - 2019-11-07 at 18 08 32](https://user-images.githubusercontent.com/1032913/68406092-dc83e480-0189-11ea-9d24-b76fd64eecbe.png) | ![Simulator Screen Shot - iPhone X - 2019-11-07 at 18 09 48](https://user-images.githubusercontent.com/1032913/68406109-e279c580-0189-11ea-8ecc-f7119952dd57.png)

hypest committed:
Remove local mock, using the global styleMock.js instead

hypest committed:
Enable testing both mobile platforms

hypest commented:
👋  @SergioEstevao , I know you've already 👍 'd the PR but, I added 0569a38 to fix the mobile tests (were failing on the gutenberg-mobile PR) and 973c2f8 to remove the hardcoded platform set and enable testing both platform so, can you have another pass please? Thanks!

SergioEstevao commented:
@SergioEstevao I had another look and all still looking good!
